src/Compiler/Compiler.cpp

```diff
@@ -805,6 +812,10 @@ bool Compiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
         std::string arg = argv[i];
         if (arg == "clean") {
           compiler->PlaceOpt(Compiler::PlacementOpt::Clean);
```

Comment:
Is clean a valid option?

Reply:
Yes, why?

Reply:
The feature being added only has options for Random and DefineOrder so clean seems out of place. I'm wondering if 2 commands are accidentally getting combined here. It seems like there's the Compiler:PlacementOpt which should have Clean and None for the placement task and then I would think we want some separate setting for this pin loc assignment altho I might be misunderstanding how it all combines in the end.

Reply:
yes, two commands are combined. You are right.

Reply:
@KochynVolodymyr @skyler-rs  clean option is also tied with place command. forexample, place clean -> clean the placement test. But for the other options those run separately.

Reply:
But does you command interact w/ the place/placement command? As far as I understand it, your options are part of the pin_loc_assign_method command so I wouldn't think you'd need to modify the place command (your changes from 999-1044 in compiler.cpp). Maybe i'm missing something?

Reply:
In my understanding, there might be two ways to run place test. If pin assign method i.e., random/in_define_order are supposed to be dependent on place test. then place command should interact with these pin assign commands. As clean is just cleaning the place test which is already done. but if user wants to place either with random pin assignment or in_define_order assignment then it should be like place random/in_define_order that's why I combined both with place as well. So, In short place random/in_define_order will execute the place test with the selected option while
and for pin_loc_assign_method command, it's an Tcl interface for selecting one of the options without running place test, just like we go to Edit Settings... on the gui and select either of the option, which only sets the selected option and then we can just type place command only to run the test. 
Because if we follow the synth command, its options are also interacting with it. Just like when i run synth area, it runs the test with setting the optimization option at "Area".

Reply:
If the requirement is just to select an options and saves the selection either with tcl interface or from the gui action, then I guess only pin_loc_assign_method command is enough.

Reply:
Ok i think I'm understanding it now, we can discuss further in the call to make sure it's working as expected, but that logic sounds correct.

Reply:
Yeah Sure!